dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -211,3 +233,90 @@ func validateFromChainState(runtimeInstance parachainruntime.RuntimeInstance, po
 
 	return &candidateCommitments, persistedValidationData, isValid, nil
 }
+
+// validateFromExhaustive validates a candidate parachain block with provided parameters
+func validateFromExhaustive(validationHost parachainruntime.ValidationHost,
+	persistedValidationData parachaintypes.PersistedValidationData,
+	validationCode parachaintypes.ValidationCode,
+	candidateReceipt parachaintypes.CandidateReceipt, pov parachaintypes.PoV) (
+	*ValidationResult, error) {
+
+	validationCodeHash := validationCode.Hash()
+	// basic checks
+	validationErr, internalErr := performBasicChecks(&candidateReceipt.Descriptor, persistedValidationData.MaxPovSize,
+		pov,
+		validationCodeHash)
+	if validationErr != nil {
+		validationResult := &ValidationResult{
+			InvalidResult: validationErr,
+		}
+		return validationResult, internalErr
+	}
+
+	validationParams := parachainruntime.ValidationParameters{
+		ParentHeadData:         persistedValidationData.ParentHead,
+		BlockData:              pov.BlockData,
+		RelayParentNumber:      persistedValidationData.RelayParentNumber,
+		RelayParentStorageRoot: persistedValidationData.RelayParentStorageRoot,
+	}
+
+	validationResult, err := validationHost.ValidateBlock(validationParams)
+	if err != nil {
+		ci := ExecutionError
+		return &ValidationResult{InvalidResult: &ci}, fmt.Errorf("executing validate_block: %w", err)
```

Comment:
As I explained above, We should return either an internal error or a validation result.

Reply:
Updated